crates/fuzzing/src/generators/memory.rs

```diff
@@ -5,6 +5,75 @@ use arbitrary::{Arbitrary, Unstructured};
 use std::ops::Range;
 use wasmtime::{LinearMemory, MemoryCreator, MemoryType};
 
+/// A description of a memory config, image, etc... that can be used to test
+/// memory accesses.
+#[derive(Debug)]
+#[allow(missing_docs)]
+pub struct MemoryAccesses {
+    pub memory_config: NormalMemoryConfig,
+    pub image: HeapImage,
+    pub offset: u32,
+    pub growth: u32,
+}
+
+impl<'a> Arbitrary<'a> for MemoryAccesses {
+    fn arbitrary(u: &mut Unstructured<'a>) -> arbitrary::Result<Self> {
+        Ok(MemoryAccesses {
+            memory_config: u.arbitrary()?,
+            image: u.arbitrary()?,
+            offset: u.arbitrary()?,
+            growth: u.arbitrary()?,
```

Comment:
Does this mean that memories are going to grow an arbitrary 32-bit amount of bytes? If so should that be capped here to stay within limits on oss-fuzz?

Reply:
An arbitrary 32-bit number of pages. I figured we could just allow and ignore growth failure, but will that run afoul of oss-fuzz?

Reply:
I think so yeah, but not always. Oss-fuzz/libfuzzer I think override malloc and mainly look at the heap size through that but IIRC it also occasionally looks through vm stats and looks for things like `mmap`, so we could allocate 1G here and blow the limits on oss-fuzz when it gets around to detecting that. Given that I think we'll only want growths which are guaranteed to either succeed for fail given our memory limits.